pkg/deploymanager/config.go

```diff
@@ -1,133 +1,63 @@
 package deploymanager
 
 import (
+	"context"
 	"os"
 
-	olmclient "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1"
+	operatorv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
 	lvmv1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/kubernetes/scheme"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
-
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// InstallNamespace is the namespace lvm is installed into
+var (
+	scheme = runtime.NewScheme()
+)
+
 const InstallNamespace = "openshift-storage"
 
-// DeployManager is a util tool used by the tests
+func init() {
+	utilruntime.Must(k8sscheme.AddToScheme(scheme))
+	utilruntime.Must(lvmv1.AddToScheme(scheme))
+	utilruntime.Must(operatorv1.AddToScheme(scheme))
+	utilruntime.Must(operatorv1alpha1.AddToScheme(scheme))
+}
+
 type DeployManager struct {
-	olmClient      *olmclient.Clientset
-	k8sClient      *kubernetes.Clientset
-	crClient       crclient.Client
-	parameterCodec runtime.ParameterCodec
-	lvmClient      *rest.RESTClient
+	crClient crclient.Client
+	Ctx      context.Context
 }
 
-// GetCrClient is the function used to retrieve the controller-runtime client.
 func (t *DeployManager) GetCrClient() crclient.Client {
 	return t.crClient
 }
 
-// GetK8sClient is the function used to retrieve the kubernetes client.
-func (t *DeployManager) GetK8sClient() *kubernetes.Clientset {
-	return t.k8sClient
-}
-
-// GetParameterCodec is the function used to retrieve the parameterCodec.
-func (t *DeployManager) GetParameterCodec() runtime.ParameterCodec {
-	return t.parameterCodec
-}
-
-// GetLvmClient is the function used to retrieve the lvm client.
-func (t *DeployManager) GetLvmClient() *rest.RESTClient {
-	return t.lvmClient
-}
-
-func getKubeconfig(kubeconfig string) (*rest.Config, error) {
```

Comment:
You can leave this function for better readibility.

Reply:
means to use the separate function for geting kubeconfig?

Reply:
Yes

Reply:
Updated that.